code/modules/antagonists/eldritch_cult/eldritch_antag.dm

```diff
@@ -0,0 +1,255 @@
+		var/count = 1
+		for(var/datum/objective/objective in objectives)
+			if(objective.check_completion())
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='greentext'>Success!</b></span>"
+			else
+				parts += "<b>Objective #[count]</b>: [objective.explanation_text] <span class='redtext'>Fail.</span>"
+				cultiewin = FALSE
+			count++
+
+	if(cultiewin)
+		parts += "<span class='greentext'>The heretic was successful!</span>"
+	else
+		parts += "<span class='redtext'>The heretic has failed.</span>"
+
+	parts += "<b>Knowledge Researched:</b> "
+
+	var/list/knowledge_message = list()
+	for(var/X in get_all_knowledge())
+		var/datum/eldritch_knowledge/EK = X
+		knowledge_message += "[EK.name]"
+	parts += knowledge_message.Join(", ")
+
+	return parts.Join("<br>")
+////////////////
+// Knowledge //
+////////////////
+
+/datum/antagonist/e_cult/proc/gain_knowledge(datum/eldritch_knowledge/EK)
+	if(has_knowledge(EK))
+		return FALSE
+	var/datum/eldritch_knowledge/initialized_knowledge = new EK
+	researched_knowledge += initialized_knowledge
+	initialized_knowledge.on_gain(owner.current)
+	return TRUE
+
+/datum/antagonist/e_cult/proc/get_researchable_knowledge()
+	var/list/researchable_knowledge = list()
+	var/list/banned_knowledge = list()
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = X
+		researchable_knowledge |= EK.next_knowledge
+		banned_knowledge |= EK.banned_knowledge
+		banned_knowledge |= EK.type
+	researchable_knowledge -= banned_knowledge
+	return researchable_knowledge
+
+/datum/antagonist/e_cult/proc/has_knowledge(datum/eldritch_knowledge/wanted)
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/searched = X
+		if(initial(wanted.name) == searched.name)
+			return TRUE
+	return FALSE
+
+/datum/antagonist/e_cult/proc/get_knowledge(datum/eldritch_knowledge/wanted)
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/searched = X
+		if(istype(searched,wanted))
+			return searched
+
+/datum/antagonist/e_cult/proc/get_all_knowledge()
+	return researched_knowledge
+
+////////////////
+// Objectives //
+////////////////
+
+/datum/objective/stalk
+	name = "spendtime"
+	var/timer = 1800 //5 minutes
+
+/datum/objective/stalk/update_explanation_text()
+	if(timer == initial(timer))//just so admins can mess with it
+		timer += pick(-600, 600)
+	if(target && target.current)
+		explanation_text = "Stalk [target.name] for at least [DisplayTimeText(timer)] while they're alive."
+	else
+		explanation_text = "Free Objective"
+
+/datum/objective/stalk/check_completion()
+	return timer <= 0 || explanation_text == "Free Objective"
+
+/datum/objective/sacrifice_ecult
+	name = "sacrifice"
+
+/datum/objective/sacrifice_ecult/update_explanation_text()
+	. = ..()
+	target_amount = rand(4,6)
+	explanation_text = "Sacrifice at least [target_amount] people."
+
+/datum/objective/sacrifice_ecult/check_completion()
+	var/list/datum/mind/owners = get_owners()
+	var/sacrificed = 0
```

Comment:
Can you keep the number of sacrificed as a variable on the datum instead, instead of compiling the number each time?